pkg/3scale/amp/component/backend.go

```diff
@@ -387,6 +398,10 @@ func (backend *Backend) buildBackendWorkerEnv() []v1.EnvVar {
 	result = append(result,
 		helper.EnvVarFromSecret("CONFIG_EVENTS_HOOK", "system-events-hook", "URL"),
 		helper.EnvVarFromSecret("CONFIG_EVENTS_HOOK_SHARED_SECRET", "system-events-hook", "PASSWORD"),
+		// TODO: Add as EnvVar or reference to CM?
```

Comment:
It's one of the usual issues that we come up with. As always, it has it's advantages and disadvantages and there doesn't seem to be a real "winner".

If you use Environment variable you don't have to worry about the ConfigMap value, querying it etc... but you have duplicated values and it's harder for the user to keep track of what's set.

If you use a ConfigMap you have to worry about the ConfigMap, but you have a centralized value for all possible references and the user can keep track of the set value for all of the objects that reference it. You also have the downside of that changing the ConfigMap value does not automatically trigger a redeployment of the elements that reference it so it might be an odd behaviour too.

Reply:
If there is no clear winner, let them be env vars then. KISS.